sage-ios/SAGE/SAGE/AnnouncementsViewController.swift

```diff
@@ -121,9 +140,55 @@ class AnnouncementsViewController: UITableViewController {
         }
         self.navigationController?.pushViewController(addAnnouncementController, animated: true)
     }
-    
-    func getAnnouncements() {
-        AnnouncementsOperations.loadAnnouncements({ (announcements) -> Void in
+
+    func showFilterOptions() {
+        let menuController = MenuController(title: "Filter Options")
+        menuController.addMenuItem(MenuItem(title: "All", handler: { (_) -> Void in
+            self.filter = nil
+            self.getAnnouncements(reset: true)
+            self.titleView.setSubtitle("All")
+        }))
+
+        if LoginOperations.getUser()!.isDirector() {
+            menuController.addMenuItem(MenuItem(title: "My School", handler: { (_) -> Void in
+                self.filter = [AnnouncementConstants.kSchoolID: String(LoginOperations.getUser()!.directorID)]
```

Comment:
isn't director id nil if the user is not a director?
so you'd want to do LoginOperations.getSchool()!.id instead